content/source/docs/enterprise/run/install-software.html.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: "enterprise2"
+page_title: "Installing Software - Runs - Terraform Enterprise"
+sidebar_current: "docs-enterprise2-run-install"
+---
+
+# Installing Custom Software
+
+In some cases, it may be useful to install certain software on the Terraform worker, 
+such as a configuration management tool or cloud CLI.
+
+Terraform's `local-exec` [provisioner](https://www.terraform.io/docs/provisioners/local-exec.html) can be used in Terraform Enterprise (TFE). This will execute commands on the host machine running Terraform. In general, it should be used in conjunction with the resource(s) the software will be used to interact with, as relying on a placeholder (such as a null resource) may result in inconsistencies as to whether the software is installed.
+
+```hcl
+resource "aws_instance" "example" {
+  ami           = "${var.ami}"
+  instance_type = "t2.micro"
+  provisioner "local-exec" {
+    command = <<EOH
+sudo apt-get update
+sudo apt-get install -y ansible
+EOH
+  }
+}
+```
+
+An [explicit `depends_on`](/intro/getting-started/dependencies.html#implicit-and-explicit-dependencies) may sometimes be needed to place the installation correctly in the provisioning sequence.
+
+Using `local-exec` is only recommended where using a more standard provider-based action is not possible or feasible. See [Terraform Runs: Custom and Community Providers](index.html#custom-and-community-providers) in the documentation for information on how to use custom and community providers in TFE.
+
+Please note that the machines that run Terraform (containers, in Private Terraform Enterprise) exist in an isolated environment and are destroyed after each use. Very little is pre-installed and nothing is persisted between Terraform runs, so you will need to install any custom software on each run.
```

Comment:
Append

> For this reason, a null resource with the `local-exec` provisioner will only install software on the first run when the `null_resource` is created. The `null_resource` will not be automatically recreated in subsequent runs, so the software will not be reinstalled into the new container, which may cause runs to encounter errors.

> However, it may be appropriate to use a null resource with a `local-exec` provisioner to install software if the null resource can be properly configured with [triggers](https://www.terraform.io/docs/provisioners/null_resource.html#example-usage).